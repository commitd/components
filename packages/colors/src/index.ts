export { presetColors } from './preset'
export type { Color } from './preset'
