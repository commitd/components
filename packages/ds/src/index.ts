// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
