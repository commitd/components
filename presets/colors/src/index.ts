export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
