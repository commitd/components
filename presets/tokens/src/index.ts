export { preset as presetTokens } from './preset'
