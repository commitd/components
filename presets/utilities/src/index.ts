export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

