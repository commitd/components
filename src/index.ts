export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
