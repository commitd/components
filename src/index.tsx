export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
