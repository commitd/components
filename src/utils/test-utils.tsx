/* eslint-disable @typescript-eslint/no-explicit-any */
/**
 * Import setupTests for your unit tests and you can use `renderLight` and `renderDark` to render elements wrapped in a ThemeProvider.
 * To render without a theme provider use `renderPlain`.
 *
 * N.B. This adds a simple custom id generator so ids match between snapshots.
 *
 */
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import './domrect-polyfill'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { ThemeProvider } from '..'
import ResizeObserver from 'resize-observer-polyfill'

// Use the polyfill for the ResizeObserver.
// This is used in some components.
global.ResizeObserver = ResizeObserver

const LightTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="light">{children}</ThemeProvider>
)

const DarkTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="dark">{children}</ThemeProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

// re-export everything
export * from '@testing-library/react'
export { userEvent }
