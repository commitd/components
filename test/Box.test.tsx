import React from 'react'
import { render } from './test-utils'
import { Box } from '../src'

describe('Box', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Box />)
    expect(asFragment()).toMatchSnapshot()
  })
})
