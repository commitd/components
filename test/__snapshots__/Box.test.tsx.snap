// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-1 Styled(MuiBox)-root-2"
  />
</DocumentFragment>
`;
